adding lifecycle_policy config block to efs_file_system (fixes #7623) (#9250)
terraform-providers/terraform-provider-aws: website/docs/r/efs_file_system.html.markdown

}
```

### Using lifecycle policy

```hcl
resource "aws_efs_file_system" "foo_with_lifecyle_policy" {
  creation_token = "my-product"

>   lifecycle_policy = {

Reviewer: This new attribute is a configuration block (TypeList + Elem: Resource), so the equals sign must be omitted in Terraform 0.12 and later.

```suggestion
  lifecycle_policy {
```

(http://docs.aws.amazon.com/efs/latest/ug/) user guide for more information.
* `encrypted` - (Optional) If true, the disk will be encrypted.
* `kms_key_id` - (Optional) The ARN for the KMS encryption key. When specifying kms_key_id, encrypted needs to be set to true.
* `lifecycle_policy` - (Optional) A file system [lifecycle policy](https://docs.aws.amazon.com/efs/latest/ug/API_LifecyclePolicy.html) object (documented below).
* `performance_mode` - (Optional) The file system performance mode. Can be either `"generalPurpose"` or `"maxIO"` (Default: `"generalPurpose"`).
* `provisioned_throughput_in_mibps` - (Optional) The throughput, measured in MiB/s, that you want to provision for the file system. Only applicable with `throughput_mode` set to `provisioned`.
* `tags` - (Optional) A mapping of tags to assign to the file system.
* `throughput_mode` - (Optional) Throughput mode for the file system. Defaults to `bursting`. Valid values: `bursting`, `provisioned`. When using `provisioned`, also set `provisioned_throughput_in_mibps`.

### Lifecycle Policy Arguments
For **lifecycle_policy** the following attributes are supported:

> * `transition_to_ia` - (Required) Indicates how long it takes to transition files to the IA storage class. Currently, the only valid value is `AFTER_30_DAYS`.

Reviewer: The schema currently defines this argument as `Optional: true` and we'll want to include the additional supported values

```suggestion
* `transition_to_ia` - (Optional) Indicates how long it takes to transition files to the IA storage class. Valid values: `AFTER_14_DAYS`, `AFTER_30_DAYS`, `AFTER_60_DAYS`, or `AFTER_90_DAYS`.
```
